src/openfermion/ops/_symbolic_operator.py

```diff
@@ -163,8 +167,12 @@ def _long_string_init(self, long_string, coefficient):
                     else:
                         coef = float(coef_string)
                 except ValueError:
-                    raise ValueError(
+                    try:
+                        coef = sympy.sympify(coef_string)
```

Comment:
I don't see this functionality tested in any of the tests. I'm actually a bit wary of allowing this, since `sympify` uses `eval` and hence this could execute arbitrary Python code. So unless there is a really good reason to support this use case, I say we don't support it.

Reply:
So the main place this would be useful is with saving and loading a SymbolicOperator, as it allows storage via plaintext. Marshal doesn't seem to work with sympy - when we use Marshal we currently cast everything to complex (which definitely doesn't work), but if I lift this requirement then I see a 'ValueError: unmarshallable object' when I try to save a sympy expression, so I think it's not supported.

Looking around, it seems sympy offers support for pickle, which might be an alternative, but apparently pickling has similar code vulnerabilities (so does marshal, it seems), so I'm not sure what to do. I'm happy to just drop support for saving operators containing sympy because there's probably no safe way to load them, or we can maybe throw in a default-off flag for the user, or perhaps something else?

Reply:
I'd rather not introduce this vulnerability into the loading of our objects, so I say we drop this feature. A better solution is to implement a more structured serialization format, like the JSON serialization in Cirq. Of course, that is a separate matter.

Reply:
SGTM, I'll drop it.

Reply:
Dropped